src/region_growing/label_connected_comp.py

```diff
@@ -5,30 +5,50 @@
 import pycc
 import cccorelib
 
+from .abstract import AbstractRegionGrowing
 
-class LabelConnectedComp:
+
+class LabelConnectedComp(AbstractRegionGrowing):
     """
     Clustering based region growing implementation using label connected comp.
     """
-    def __init__(self, octree_level=9, min_component_size=100):
+    def __init__(self, label, exclude_labels, octree_level=9,
+                 min_component_size=100):
+        super().__init__(label)
         """ Init variables. """
         self.octree_level = octree_level
         self.min_component_size = min_component_size
 
-    def set_input_cloud(self, las, mask):
+        self.exclude_labels = exclude_labels
+        self.seed_point_label = label
+
+    def _set_mask(self, las_labels):
+        """ Configure the points that we want to perform region growing on. """
+        mask = np.full(len(las_labels), True)
+
+        for exclude_label in self.exclude_labels:
+            mask = mask & (las_labels != exclude_label)
+
+        self.las_label = las_labels
+        self.mask = mask
+
+    def _convert_input_cloud(self, las, header):
         """ Function to convert to CloudCompare point cloud. """
         # Be aware that CloudCompare stores coordinates on 32 bit floats.
         # To avoid losing too much precision you should 'shift' your
         # coordinates if they are 64 bit floats (which is the default in
         # python land)
-        xs = (las.x[mask] - las.header.x_min).astype(pycc.PointCoordinateType)
-        ys = (las.y[mask] - las.header.y_min).astype(pycc.PointCoordinateType)
-        zs = (las.z[mask] - las.header.z_min).astype(pycc.PointCoordinateType)
+        xs = (las['x'][self.mask] -
```

Comment:
points is nu geen dict meer, dus `points[self.mask, 0]` ipv `points['x'][self.mask]`